src/NuGet.Indexing/DownloadsByVersion.cs

```diff
@@ -48,8 +41,13 @@ public int Total
 
             set
             {
+                if (_downloadsByVersion.ContainsKey(version))
+                {
+                    _total = _total - _downloadsByVersion[version];
```

Comment:
Because you are operating on the live index, regardless of how you write this method, you will always have the download counts and total out of sync for a single operation. This has little impact, but could create some absolutely nasty issues in the future.

To prevent this, the entire `DownloadsByVersion` object should be constructed in memory and then swapped with the instance contained in the dictionaries in `Downloads`. We should adopt a similar flow for the other auxiliary data loads as well.